File: structures-js/structures-e2e/test/services/ClusterInfoService.ts
import { Continuum, ContinuumSingleton, IServiceProxy } from "@kinotic/continuum-client";
import { ClusterInfo } from "../domain/ClusterInfo";

export interface ClusterInfoService {

    getClusterInfo(): Promise<ClusterInfo>;

}

export class DefaultClusterInfoService implements ClusterInfoService {

    protected serviceProxy: IServiceProxy

    constructor(continuumSingleton?: ContinuumSingleton) {
        this.serviceProxy = (continuumSingleton ? continuumSingleton : Continuum).serviceProxy('org.kinotic.structures.api.services.cluster.ClusterInfoService')
    }

    getClusterInfo(): Promise<ClusterInfo> {
        return this.serviceProxy.invoke('getClusterInfo');
    }

}

File: structures-core/src/main/java/org/mindignited/structures/api/services/cluster/ClusterInfoService.java
package org.mindignited.structures.api.services.cluster;

// import org.mindignited.continuum.api.annotations.Publish;
// import org.mindignited.continuum.api.annotations.Version;
import org.mindignited.structures.api.domain.cluster.ClusterInfo;

import reactor.core.publisher.Mono;

/**
 * Provides information about the ignite structures cluster.
 */
// @Publish
// @Version("1.0.0")
public interface ClusterInfoService {
    
    /**
     * Returns the information about the ignite structures cluster.
     * 
     * @return the information about the ignite structures cluster
     */
    Mono<ClusterInfo> getClusterInfo();

}
